uaclient/log.py

```diff
@@ -58,3 +62,35 @@ def format(self, record: logging.LogRecord) -> str:
 
         local_log_record["extra"] = extra_message_dict
         return json.dumps(list(local_log_record.values()))
+
+
+def get_user_log_file() -> str:
+    """Gets the correct user log_file storage location
+
+    returns XDG_CACHE_HOME env variable, if it is set,
+    If not set, we return the default path
+    """
+    user, _ = system.subp(["whoami"])
+    user = user.strip()
+    user_cache = os.environ.get("XDG_CACHE_HOME")
+    log_file = "{}-{}.log".format(defaults.USER_LOG_FILE_PATH, user)
+    if user_cache:
+        return (pathlib.Path(user_cache) / log_file).as_posix()
+    return (pathlib.Path.home() / ".cache" / log_file).as_posix()
+
+
+def get_users_log_file() -> List[str]:
```

Comment:
recommended name change
```suggestion
def get_all_user_log_files() -> List[str]:
```